eZ/Publish/Core/Persistence/Legacy/User/Handler.php

```diff
@@ -279,6 +355,38 @@ public function deleteRole($roleId)
     }
 
     /**
+     * Publish the specified role draft.
+     *
+     * @param mixed $roleId
+     */
+    public function publishRoleDraft($roleId)
+    {
+        $roleDraft = $this->loadRoleDraft($roleId);
+
+        $this->roleGateway->publishRoleDraft($roleDraft->id);
```

Comment:
What about publishing a draft from an existing role? This will probably fail with a DB exception as the gateway just updates the role record, so then we end up with 2 records with same PKs.
See [how it is done in `Content\Type\Handler`](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Persistence/Legacy/Content/Type/Handler.php#L569-L582). We first delete the old one and then publish the new one.